src/file_organizer/pipeline/resource_aware_executor.py

```
@@ -0,0 +1,421 @@
+        """Return ``(ctx, buffer, start_time)`` for file *idx*.
+
+        Uses a completed prefetch future when available; falls back to
+        inline I/O. Any exception from either path is recorded on a
+        fresh failed context so the batch loop keeps moving.
+        """
+        if idx in futures:
+            future, start_time = futures.pop(idx)
+            try:
+                ctx, buffer = future.result()
+                return ctx, buffer, start_time
+            except Exception as exc:  # Intentional catch-all: future can raise any stage error
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)